### ISSUE TITLE
Pass a random_state argument for reproducibility when using random variables

### ISSUE THREAD

**Issue by guilgautier** [For-a-few-DPPs-more/structure-factor#3]
It would be great to be able to pass a `random_state` or `seed` or `rng` argument for reproducibility when calling function involving random variables.
See also the `check_random_state` function in sklearn
https://github.com/scikit-learn/scikit-learn/blob/a45c0c99a38cffca6724cb8fd38b12edd4fb6b35/sklearn/utils/validation.py#L926
